src/middlewared/middlewared/plugins/ldap.py

```diff
@@ -918,22 +921,21 @@ async def __start(self, job, ds_type):
                 ds_type = DSType.LDAP
                 await self.middleware.call('directoryservices.health.set_state', ds_type.value, DSStatus.JOINING.name)
 
-        cache_job_id = await self.middleware.call('directoryservices.connection.activate')
+        if dom_join_resp == DomainJoinResponse.ALREADY_JOINED.name:
+            cache_job_id = await self.middleware.call('directoryservices.connection.activate')
+            await job.wrap(await self.middleware.call('core.job_wait', cache_job_id))
 
-        await job.wrap(await self.middleware.call('core.job_wait', cache_job_id))
         await self.middleware.call('directoryservices.health.set_state', ds_type.value, DSStatus.HEALTHY.name)
```

Comment:
The `set_state` to `HEALTHY` here doesn't seem correct.  We _could_ have just set it to JOINING above.

_Maybe_ indent under
```
if dom_join_resp in (DomainJoinResponse.ALREADY_JOINED.value, DomainJoinResponse.PERFORMED_JOIN.value):
```
line?

(Actually now I'm wondering whether we need this same condition above too.)

Reply:
I'll add comments.